database/src/test/java/database/service/SqlDatabaseServiceTest.java

```diff
@@ -0,0 +1,274 @@
+import database.exception.IdProvidedManuallyException;
+import database.exception.InvalidParameterValueException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.List;
+
+import static database.service.SqlDatabaseService.ENTITY_IS_NOT_FOUND;
+import static database.service.SqlDatabaseService.ID_PROVIDED_MANUALLY;
+import static database.service.SqlDatabaseService.INVALID_PARAMETER_VALUE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class SqlDatabaseServiceTest {
+    private SqlDatabaseService sqlDatabaseService;
+    private Student firstStudent;
+    private Student secondStudent;
+    private Student thirdStudent;
+    private Student fourthStudent;
+    private final String DB_USER_NAME = "test_user";
+    private final String DB_PASSWORD = "Qwerty!1";
+    private final String DATABASE_NAME = "test_entities";
+    private final String DATABASE_URL = "jdbc:mysql://localhost:3306/";
+    private final String UNABLE_CREATE_TABLE =
+            "Unable to create table. Please check if it already exists: Table 'student' already exists";
+    private final String TABLE_NOT_EXIST = "Table does not exist: Student";
+    private MockedStatic<DriverManager> driverManagerMock;
+
+    @BeforeEach
+    void setUp() throws SQLException {
+        firstStudent = new Student("FirstName1 LastName1", 5.0);
+        secondStudent = new Student("FirstName2 LastName2", 4.5);
+        thirdStudent = new Student("FirstName3 LastName3", 5.0);
+        fourthStudent = new Student("FirstName1 LastName1", 5.0);
+        sqlDatabaseService = new SqlDatabaseService(DATABASE_URL, DB_USER_NAME, DB_PASSWORD, DATABASE_NAME);
+    }
+
+    @AfterEach
+    void tearDown() {
+        try {
+            sqlDatabaseService.deleteTable(Student.class);
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    @Test
+    void createTableTest() throws SQLException {
+        Class<? extends BaseEntity> entityClass = Student.class;
+
+        assertTrue(sqlDatabaseService.createTable(entityClass));
+        assertTrue(sqlDatabaseService.checkTableExists(DATABASE_NAME, "Student"));
+    }
+
+    @Test
+    void createTableFailedTest() {
+        Class<? extends BaseEntity> entityClass = Student.class;
+        sqlDatabaseService.createTable(entityClass);
+
+        CreationDatabaseException exception = assertThrows(CreationDatabaseException.class, () ->
+                sqlDatabaseService.createTable(entityClass));
+        assertEquals(UNABLE_CREATE_TABLE, exception.getMessage());
```

Comment:
Please use `UNABLE_CREATE_TABLE` constant from service class and check with `assertTrue(exception.getMessage().startswith(UNABLE_CREATE_TABLE)`).